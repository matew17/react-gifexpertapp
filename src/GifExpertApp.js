import React, { useState } from 'react'

import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

function GifExpertApp({ defaultCategories = ['Kimetsu no yaiba'] }) {
    const [categories, setCategory] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpertApp</h2>

            <AddCategory setCategory={setCategory} />

            <hr />

            <ul>
                {categories.map(category => (
                    <GifGrid
                        key={category}
                        category={category}
                    />
                ))}
            </ul>
        </>
    )
}

export default GifExpertApp
