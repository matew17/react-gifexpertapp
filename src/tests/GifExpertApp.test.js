import { shallow } from "enzyme";
import React from "react";

import GifExpertApp from './../GifExpertApp';

describe('probar GifExpertApp', () => {

    test('debe renderizar el componente ', () => {
        const wrapper = shallow(
            <GifExpertApp />
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('debe renderizar el componente ', () => {
        const defaultCats = ['One Punch', 'Kimetsu'];
        const wrapper = shallow(
            <GifExpertApp defaultCategories={defaultCats} />
        );

        expect(wrapper.find('GifGrid').length).toBe(defaultCats.length);
    });
});
