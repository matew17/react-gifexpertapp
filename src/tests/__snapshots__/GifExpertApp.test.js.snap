// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probar GifExpertApp debe renderizar el componente  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="Kimetsu no yaiba"
      key="Kimetsu no yaiba"
    />
  </ul>
</Fragment>
`;
